src/plan_expand_hypertable.c

```diff
@@ -306,12 +309,112 @@ get_chunk_oids(CollectQualCtx *ctx, PlannerInfo *root, RelOptInfo *rel, Hypertab
 																	  reverse);
 		}
 		else
-			return find_children_oids(hri, ht, AccessShareLock);
+		{
+			List *child_oids = find_children_oids(hri, ht, AccessShareLock);
+
+			/* Exclude the root parent table */
+			return list_delete_oid(child_oids, ht->main_table_relid);
+		}
 	}
 	else
 		return get_explicit_chunk_oids(ctx, ht);
 }
 
+#if !(PG96 || PG10)
+
+/*
+ * Create partition expressions for a hypertable.
+ *
+ * Build an array of partition expressions where each element is a list of
+ * expression for a particular partitioning dimension.
+ *
+ * The partition expressions are used by, e.g., group_by_has_partkey() to check
+ * whether a GROUP BY clause covers all partitioning dimesions.
+ */
+static List **
+get_hypertable_partexprs(Hypertable *ht, Index varno)
+{
+	int i;
+	List **partexprs;
+
+	Assert(NULL != ht->space);
+
+	partexprs = palloc0(sizeof(List *) * ht->space->num_dimensions);
+
+	for (i = 0; i < ht->space->num_dimensions; i++)
+	{
+		Dimension *dim = &ht->space->dimensions[i];
+		Expr *expr;
+
+		/* Only functions on open dimensions need to be function expressions */
+		if (IS_OPEN_DIMENSION(dim) && NULL != dim->partitioning)
+			expr = (Expr *) dim->partitioning->partfunc.func_fmgr.fn_expr;
+		else
+		{
+			HeapTuple tuple = SearchSysCacheAttNum(ht->main_table_relid, dim->column_attno);
+			Form_pg_attribute att;
+
+			if (!HeapTupleIsValid(tuple))
+				elog(ERROR, "cache lookup failed for attribute");
+
+			att = (Form_pg_attribute) GETSTRUCT(tuple);
+
+			expr = (Expr *) makeVar(varno,
+									dim->column_attno,
```

Comment:
Maybe check/assert that `dim->column_attno > 0`

Reply:
I'd rather put such an assert where `dim->column_attno` is set or put into the Hypertable cache. Otherwise, I am not sure why we single out this value when we could assert a lot of other things too.